src/services/timePicker.ts

```diff
@@ -0,0 +1,129 @@
+      const timeZone = getTimeZone();
+      return timeOptions.filter((timeOption) => {
+        const timeRange = rangeUtil.convertRawToRange(timeOption, timeZone);
+        if (timeRange) {
+          // This will return the exact duration for the interval, if the interval covers DST there will be an extra/missing hour!
+          const intervalSeconds = Math.floor((timeRange.to.valueOf() - timeRange.from.valueOf()) / 1000);
+          // Pad retention by 10%, there's no downside to querying over retention besides some empty space in the query, and it might be frustrating to not get a time range if retention is close
+          const retentionGreaterThanInterval = intervalSeconds <= maxRetentionSeconds * 1.1;
+          const pluginConfigGreaterThanInterval = intervalSeconds <= maxPluginConfigSeconds;
+          return intervalSeconds === 0 || retentionGreaterThanInterval || pluginConfigGreaterThanInterval;
+        }
+
+        return false;
+      });
+    }
+  }
+
+  return timeOptions;
+};
+
+// Taken from grafana-ui/src/components/DateTimePickers/options.ts and adapted for typical logs searches and retentions
+export const quickOptions: TimeOption[] = [
+  { from: 'now-1m', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-1-mins', 'Last minute') },
+  { from: 'now-5m', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-5-mins', 'Last 5 minutes') },
+  {
+    from: 'now-15m',
+    to: 'now',
+    display: t('grafana-ui.date-time-pickers.quick-options.last-15-mins', 'Last 15 minutes'),
+  },
+  {
+    from: 'now-30m',
+    to: 'now',
+    display: t('grafana-ui.date-time-pickers.quick-options.last-30-mins', 'Last 30 minutes'),
+  },
+  { from: 'now-1h', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-1-hour', 'Last 1 hour') },
+  { from: 'now-3h', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-3-hours', 'Last 3 hours') },
+  { from: 'now-6h', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-6-hours', 'Last 6 hours') },
+  {
+    from: 'now-12h',
+    to: 'now',
+    display: t('grafana-ui.date-time-pickers.quick-options.last-12-hours', 'Last 12 hours'),
+  },
+  {
+    from: 'now-24h',
+    to: 'now',
+    display: t('grafana-ui.date-time-pickers.quick-options.last-24-hours', 'Last 24 hours'),
+  },
+  { from: 'now-2d', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-2-days', 'Last 2 days') },
+  { from: 'now-7d', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-7-days', 'Last 7 days') },
+  { from: 'now-30d', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-30-days', 'Last 30 days') },
+  { from: 'now-60d', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-60-days', 'Last 60 days') },
+  { from: 'now-90d', to: 'now', display: t('grafana-ui.date-time-pickers.quick-options.last-90-days', 'Last 90 days') },
```

Comment:
LGTM only seeing the last 30 days for my retention_period:"30d"

<img width="1758" height="912" alt="Image" src="https://github.com/user-attachments/assets/cfb51f02-7a8c-4045-aa34-c890c835c121" />